### PR TITLE
feat: display OGA and submission responses in trader app

### LINKED ISSUE #161: feat: Display OGA submission response in trader portal
### Problem

In the trader portal, when a filled form is submitted to an OGA (Other Government Agency), the API response returned from that submission is not displayed to the trader. The current `SimpleForm` plugin only renders information when a callback arrives from the OGA (i.e. the `OGA_REVIEWED` / `OGA_ACKNOWLEDGED` state), but there is no UI to surface the immediate response from the OGA submission itself (the `OGA_ACKNOWLEDGED` transition triggered via `SUBMIT_FORM_AWAIT_OGA`).

This leaves the trader without feedback after submission — they cannot see acknowledgement details, reference numbers, or any other data the OGA returns synchronously at the point of submission.

### Proposed Solution

Extend the `SimpleForm` plugin (and its configuration type) to support an optional third form panel — an "OGA Submission Response" view — that is rendered when the plugin state is `OGA_ACKNOWLEDGED`. The backend already returns a response payload from the OGA submission; the frontend should:

1. Include the OGA submission response data in the `SimpleFormConfig` returned by the backend (alongside `traderFormInfo` and `ogaReviewForm`).
2. Render a read-only form or structured display panel in `SimpleForm.tsx` showing that response when `pluginState === 'OGA_ACKNOWLEDGED'`.

This mirrors how `OgaReviewForm` handles the callback data, but targets the synchronous submission response instead.

### Alternatives

- Display a generic success banner without structured data — this was deemed insufficient as the OGA response may contain reference numbers or status details relevant to the trader.
- Poll the backend for status updates — rejected in favour of surfacing the data already available in the immediate API response.

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

Adds support for displaying immediate OGA (Other Government Agency) callback responses and submission responses directly in the Trader App's SimpleForm plugin. Previously, the OGA response was only shown after a specific FSM state transition; now both submission and OGA response forms are always rendered when data is available in the local store.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (no functional changes)

## Changes Made

- **`backend/internal/task/plugin/simple_form.go`**:
  - Replaced the `attachOgaContent` method with a generic `attachFormDisplay` helper that handles both submission and OGA response form rendering.
  - Added `displayFormID` helper to safely extract form IDs from config.
  - Submission response is now written to the local store (`submissionResponse` key) on form submit so it can be displayed immediately.
  - Both `submissionResponse` and `ogaResponse` forms are attached to the render payload unconditionally when data exists in the store.

- **`backend/internal/task/container/container.go`**:
  - Streamlined plugin state retrieval in the `Start` method by using `GetPluginState()` instead of manually acquiring a read lock.
  - Minor cleanup: removed blank lines and combined variable declarations for conciseness.

- **`portals/apps/trader-app/src/plugins/SimpleForm.tsx`**:
  - Added `submissionResponseForm` to `SimpleFormConfig` type.
  - Added new `SubmissionResponseForm` component with an emerald-themed UI to display the submission response inline.
  - Restyled `OgaReviewForm` component with an indigo-themed header for visual distinction.
  - Both response forms are conditionally rendered below the main trader form when their data is present.

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Closes #161

## Screenshots/Demo
Can clearly see, the immediate response from the customs is visible here.
image
## Additional Notes

The `attachFormDisplay` refactor makes it straightforward to add more response form types in the future by calling the same helper with different store keys, form IDs, and content keys.

## Deployment Notes

No special deployment steps required.

### CODE REVIEW

File: portals/apps/trader-app/src/plugins/SimpleForm.tsx [hunk 1]
```diff
@@ -146,13 +147,51 @@ function TraderForm(props: { formInfo: TaskFormData, pluginState: string }) {
...
+        <JsonForms
+          schema={props.formInfo.schema}
+          uischema={props.formInfo.uiSchema}
+          data={props.formInfo.formData}
+          renderers={radixRenderers}
+          readonly={true}
+        />
+      </div>
+    </div>
+  )
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply portals/apps/trader-app/src/plugins/SimpleForm.tsx]: Done

**Review by sthanikan2000 — APPROVED** [review]
LGTM!

**Review by ginaxu1 — APPROVED** [review]
